metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1426,6 +1426,12 @@ class MetalsLanguageServer(
   ): CompletableFuture[Unit] = {
     val path = AbsolutePath(event.path)
     val isScalaOrJava = path.isScalaOrJava
+
+    event.eventType match {
+      case EventType.CreateOrModify if path.isBsp =>
+        quickConnectToBuildServer()
```

Comment:
I was thinking that probably it would be nice if we start generating `.bsp` for scala-cli from Metals as an option for importing build if there no other build tools were detected. Also, the same approach might be applied for sbt and mill. For example if we see that it's sbt we can suggest two options for import even if there is no `.bsp` directory.

Maybe it would be better to improve the importing logic rather than adding this handler? wdyt?
I think it will lead to duplicated reconnect calls.

Reply:
It doesn't lead to duplicated reconnection from what I checked. We only listen for .BSP only if we don't find any build tool.

I don't think we should yet propose scalacli as default(not sure if it wouldn't be flaky) and this isn't actually exclusive. People might run scalacli manually and it's sensible to connect to it in that case.

Reply:
Improving importing logic might also be good, but it's not necessarily related to this and doesn't solve the issue of what happens if BSP was generated outside of metals. People shouldn't need to restart metals in that case.

Reply:
Also added a check for bspSession to make sure it's empty.

Reply:
Ach actually that caused issues with tests. I think it's fine to reconnect whenever `.bsp` changes since it need to be changed explicitely.